cmd/bootstrap/run/qc.go

```diff
@@ -45,11 +45,17 @@ func (pd *ParticipantData) Identities() flow.IdentityList {
 // which assumes that only consensus participants construct QCs, which also have produce votes.
 //
 // TODO: modularize QC construction code (and code to verify QC) to be instantiated without needing private keys.
-func GenerateRootQC(block *flow.Block, votes []*model.Vote, participantData *ParticipantData, identities flow.IdentityList) (*flow.QuorumCertificate, error) {
+// It returns (qc, nil, nil) if a QC can be constructed with enough votes, and there is no invalid votes
+// It returns (qc, invalidVotes, nil) if there are some invalid votes, but a QC can still be constructed
+// It returns (nil, invalidVotes, err) if no qc can be constructed with not enough votes or running any any exception
+func GenerateRootQC(block *flow.Block, votes []*model.Vote, participantData *ParticipantData, identities flow.IdentityList) (
+	*flow.QuorumCertificate, // the constructed QC
+	[]error, // return invalid votes error
+	error) { // exception or could not construct qc
```

Comment:
```suggestion
	error,  // exception or could not construct qc
) {
```
nitpick: I find these much easier to read when there is a more visible distinction between the return variable line and the first line of the function.